go-controller/pkg/ovn/logical_switch_cache.go

```diff
@@ -0,0 +1,148 @@
+package ovn
+
+import (
+	"fmt"
+	"net"
+	"sync"
+
+	ipam "k8s.io/kubernetes/pkg/registry/core/service/ipallocator"
+
+	"reflect"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	"k8s.io/klog"
+)
+
+// logicalSwitchInfo contains information corresponding to the node. It holds the
+// subnet allocations (v4 and v6) as well as the IPAM allocator instances for each
+// subnet managed for this node
+// logicalSwitchInfo is used to save switch specific information such as subnets,
+// allocated IPs per switch etc.
+type logicalSwitchInfo struct {
+	hostSubnets []*net.IPNet
+	ipams       []ipam.Interface
+}
+
+type logicalSwitchCache struct {
+	cache map[string]logicalSwitchInfo
+	// A RW mutex for logicalSwitchCache which holds logicalSwitch information
+	sync.RWMutex
+}
+
+func (cache *logicalSwitchCache) AddToLogicalSwitchCache(nodeName string, hostSubnets []*net.IPNet) error {
```

Comment:
AddToLogicalSwitchCache seems redundant; it's a method on the logical switch cache object already, so we should name the logical switch cache object appropriately. eg this would be pretty redundant:

`ovn.logicalSwitchCache.AddToLogicalSwitchCache()`

Reply:
Ok, will rename the method to `Add`